e2e/tcproute_test.go

```diff
@@ -441,3 +443,231 @@ func defaultNetworkURL() string {
+	return []*networkservices.TcpRouteRouteRule{
+		{
+			Action: &networkservices.TcpRouteRouteAction{
+				Destinations: []*networkservices.TcpRouteRouteDestination{
+					{
+						ServiceName: resourceSelfLink(bs1ID),
+						Weight:      10,
+					},
+				},
+			},
+			Matches: []*networkservices.TcpRouteRouteMatch{
+				{
+					Address: routeCIDR,
+					Port:    "80",
+				},
+			},
+		},
+		{
+			Action: &networkservices.TcpRouteRouteAction{
+				Destinations: []*networkservices.TcpRouteRouteDestination{
+					{
+						ServiceName: resourceSelfLink(bs2ID),
+						Weight:      10,
+					},
+				},
+			},
+			Matches: []*networkservices.TcpRouteRouteMatch{
+				{
+					Address: routeCIDR2,
+					Port:    "80",
+				},
+			},
+		},
+	}
+}
+
+func createTCPRoutes(t *testing.T, graphBuilder *rgraph.Builder, numTCPR int, name_prefix string, meshURL string, bss []*cloud.ResourceID) ([]*cloud.ResourceID, error) {
+	var tcprs []*cloud.ResourceID
+	var e error
+	for i := 0; i < 2*numTCPR; i += 2 {
+		tcprRules1 := createTcprRules(bss[i], "10.240."+strconv.Itoa(i)+".83/32", bss[i+1], "10.240."+strconv.Itoa(i+1)+".83/32")
+		name := name_prefix + "-" + strconv.Itoa(i/2)
+		tcpr, err := buildTCPRoute(graphBuilder, name, meshURL, tcprRules1, bss[i])
+		if err != nil {
+			errors.Join(e, fmt.Errorf("buildTcpRoute(_, %s, %s, %v, %s) = %v, want nil", name, meshURL, tcprRules1, bss[i], err))
+		}
+		tcprs = append(tcprs, tcpr)
+		t.Logf("%s = %s", name, pretty.Sprint(tcpr))
+	}
+	return tcprs, e
+}
+func TestMeshWithMultipleTCPRoutes(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	graphBuilder := rgraph.NewBuilder()
+	res_infix := "multiple-tcpr"
+	meshURL, meshKey := ensureMesh(ctx, t, "multiple-tcpr-mesh")
+
+	t.Cleanup(func() {
+		err := theCloud.Meshes().Delete(ctx, meshKey)
+		t.Logf("theCloud.Meshes().Delete(ctx, %s): %v", meshKey, err)
+	})
+
+	hcs, err := createManyHealthchecks(graphBuilder, 6, res_infix)
+	if err != nil {
+		t.Fatalf("createManyHealthchecks(_, 6, %s) = (_, %v), want (_, nil)", res_infix, err)
+	}
+
+	bss, err := createManyBackendServicesWithHC(graphBuilder, 6, res_infix, hcs)
+	if err != nil {
+		t.Fatalf("createManyBackendServicesWithHC(_, 6, %s) = (_, %v), want (_, nil)", res_infix, err)
+	}
+
+	tcprs, err := createTCPRoutes(t, graphBuilder, 3, res_infix, meshURL, bss)
+
+	graph, err := graphBuilder.Build()
+	if err != nil {
+		t.Fatalf("graphBuilder.Build() = %v, want nil", err)
+	}
+
+	result, err := plan.Do(ctx, theCloud, graph)
+	if err != nil {
+		t.Fatalf("plan.Do(_, _, _) = %v, want nil", err)
+	}
+
+	ex, err := exec.NewSerialExecutor(result.Actions)
+	if err != nil {
+		t.Fatalf("exec.NewSerialExecutor(_, _) err: %v", err)
+		return
+	}
+	res, err := ex.Run(ctx, theCloud)
+	if err != nil || res == nil {
+		t.Errorf("ex.Run(_,_) = %v, want nil", err)
+	}
+
+	t.Cleanup(func() {
```

Comment:
I think this clean up is to late. What if you hit error in line 557. Then all resources will leak